src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
```

Comment:
Use a `switch/case` structure instead of multiple `if`s against the same variable with well defined values